publisher-sdk-tests/src/androidTest/java/com/criteo/publisher/integration/StandaloneFunctionalTest.java

```diff
@@ -108,6 +108,7 @@ public void setUp() throws Exception {
 
     api = spy(dependencyProvider.providePubSdkApi());
     when(dependencyProvider.providePubSdkApi()).thenReturn(api);
+    when(androidUtil.isCurrentProcessInForeground(any())).thenReturn(true);
```

Comment:
Nit: you can transform the `@Mock` to `@SpyBean` to make it instantiated via the mocked dependency provider.
Honestly, I don't know if it work if you use `@MockBean`. As the annotations are processed after your code in `MockedDependencyProvider`, the annotations processor might not copy the previous stub registered.